plugins/inputs/mysql/mysql.go

```diff
@@ -1227,6 +1274,142 @@ func (m *Mysql) gatherInnoDBMetrics(db *sql.DB, serv string, acc telegraf.Accumu
 	return nil
 }
 
+// gatherPerfSummaryPerAccountPerEvent can be used to fetch enabled metrics from
+// performance_schema.events_statements_summary_by_account_by_event_name
+func (m *Mysql) gatherPerfSummaryPerAccountPerEvent(db *sql.DB, serv string, acc telegraf.Accumulator) error {
+
+	sqlQuery := perfSummaryPerAccountPerEvent
+
+	var rows *sql.Rows
+	var err error
+
+	var (
+		srcUser                 string
+		srcHost                 string
+		eventName               string
+		countStar               float64
+		sumTimerWait            float64
+		minTimerWait            float64
+		avgTimerWait            float64
+		maxTimerWait            float64
+		sumLockTime             float64
+		sumErrors               float64
+		sumWarnings             float64
+		sumRowsAffected         float64
+		sumRowsSent             float64
+		sumRowsExamined         float64
+		sumCreatedTmpDiskTables float64
+		sumCreatedTmpTables     float64
+		sumSelectFullJoin       float64
+		sumSelectFullRangeJoin  float64
+		sumSelectRange          float64
+		sumSelectRangeCheck     float64
+		sumSelectScan           float64
+		sumSortMergePasses      float64
+		sumSortRange            float64
+		sumSortRows             float64
+		sumSortScan             float64
+		sumNoIndexUsed          float64
+		sumNoGoodIndexUsed      float64
+	)
+
+	var events []interface{}
+	if len(m.PerfSummaryEvents) > 0 {
+		sqlQuery += " WHERE EVENT_NAME IN ("
+		for i, eventName := range m.PerfSummaryEvents {
+			if i > 0 {
+				sqlQuery += ", "
+			}
+			sqlQuery += "?"
+			events = append(events, eventName)
+		}
+		sqlQuery += ")"
+
+		rows, err = db.Query(sqlQuery, events...)
+	} else {
+		rows, err = db.Query(perfSummaryPerAccountPerEvent)
+	}
+
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	// parse DSN and save server tag
+	servtag := getDSNTag(serv)
+	tags := map[string]string{"server": servtag}
+	for rows.Next() {
+		if err := rows.Scan(
+			&srcUser,
+			&srcHost,
+			&eventName,
+			&countStar,
+			&sumTimerWait,
+			&minTimerWait,
+			&avgTimerWait,
+			&maxTimerWait,
+			&sumLockTime,
+			&sumErrors,
+			&sumWarnings,
+			&sumRowsAffected,
+			&sumRowsSent,
+			&sumRowsExamined,
+			&sumCreatedTmpDiskTables,
+			&sumCreatedTmpTables,
+			&sumSelectFullJoin,
+			&sumSelectFullRangeJoin,
+			&sumSelectRange,
+			&sumSelectRangeCheck,
+			&sumSelectScan,
+			&sumSortMergePasses,
+			&sumSortRange,
+			&sumSortRows,
+			&sumSortScan,
+			&sumNoIndexUsed,
+			&sumNoGoodIndexUsed,
```

Comment:
Hmm I think this won't work if you specify only a subset of events using the `perf_summary_events` option. So either you change the logic here or remove the option to select the events to query...

Reply:
I think here is a bit misunderstanding.
Main source of data is table: performance_schema.events_statements_summary_by_account_by_event_name

table structure is:
event_name | metrics | which | are | provided | above.

Basically saying, it's key-value structure. Table has many "keys". Providing  `perf_summary_events` just limit number of rows to be collected, basically saying we filter rows, but colunms will be always the same.

Reply:
Oh I see, so it limits what comes out in the third column and thus the rows, but the row structure stays the same!? 

If so, then please mark this part as resolved. But please consider putting this explanation as a comment next to the code above! Just for the next one not asking the same question... :-)

Reply:
> but the row structure stays the same!?
Yes, exactly!

I will add comment. Thank you.